lib/acunetix/scan.rb

```diff
@@ -45,30 +45,57 @@ def method_missing(method, *args)
       # The problem would be that it would make tricky to debug problems with
       # typos. For instance: <>.potr would return nil instead of raising an
       # exception
-      unless supported_tags.include?(method)
-        super
-        return
+      super and return unless SUPPORTED_TAGS.include?(method)
+
+      if tag = xml.at_xpath("./#{tag_name_for_method(method)}")
+        tag.text
+      else
+        nil
       end
+    end
+
+
+    def report_items
+      @xml.xpath('./ReportItems/ReportItem')
+    end
+
+
+    def service
+      "port #{start_url_port}, #{banner}"
+    end
+
 
-      # first we try the attributes: name
-      # translations_table = {}
-      # method_name = translations_table.fetch(method, method.to_s)
-      # return @xml.attributes[method_name].value if @xml.attributes.key?(method_name)
+    def start_url_host
+      start_uri.host
+    end
+    alias_method :hostname, :start_url_host
+
+
+    def start_url_port
+      start_uri.port
+    end
+
+
+    def to_node(content_service)
+      node = content_service.create_node(label: start_url_host, type: :host)
+      node.set_property(:hostname, hostname)
+      node.set_property(:service,  service)
```

Comment:
The property should be named `services`, and the content of the service should be splitted into meaningful parts (such as port, protocol, product, version, etc) so the node properties engine can build a table and keep track of the different services properly. Take a look at the implementation in dradis-nmap: https://github.com/dradis/dradis-nmap/blob/master/lib/dradis/plugins/nmap/importer.rb#L47